chalk-solve/src/clauses/builtin_traits/fn_family.rs

```diff
@@ -110,6 +110,21 @@ pub fn add_fn_trait_program_clauses<I: Interner>(
             Ok(())
         }
         TyKind::Closure(closure_id, substitution) => {
+            for custom in db.custom_clauses() {
+                if let DomainGoal::LocalImplAllowed(tr) =
```

Comment:
Wait, why do we have to add this `LocalImplAllowed`? I'm fine with funky testing frameworks, but this seems to be a bit more invasive to the "lower layers" than I would like.